dotCMS/src/main/java/com/dotcms/storage/StorageKey.java

```diff
@@ -0,0 +1,70 @@
+package com.dotcms.storage;
+
+/**
+ * Encapsulates a storage key, it has the key (for instance the path)
+ * the group (for instance the bucket, folder, space, etc)
+ * Storage {@link StorageType}
+ * @author jsanca
+ */
+public class StorageKey {
+
+    private final String key;
+    private final String group;
+    private final String storage;
+
+    private StorageKey(final Builder builder) {
+
+        this.key = builder.key;
+        this.group = builder.group;
+        this.storage    = builder.storage;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    public String getGroup() {
+        return group;
+    }
+
+    public String getStorage() {
+        return storage;
+    }
+
+    public static final class Builder {
+
+        private String key;
```

Comment:
Codacy found an issue: [Field key has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)